aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,498 @@
+    use aptos_token::token::{Self, TokenId};
+    use aptos_framework::coin;
+    use aptos_framework::event::{Self, EventHandle};
+    use std::string::String;
+    use aptos_framework::timestamp;
+    use aptos_framework::account;
+    use aptos_std::guid::{Self, ID};
+    use std::error;
+
+
+    const EINVALIDE_EXPIRATION_TIME: u64 = 1;
+    const EINVALIDE_START_TIME: u64 = 2;
+    const EAUCTION_NOT_EXIST: u64 = 3;
+    const EBID_INCREASE_TOO_SMALL: u64 = 4;
+    const EAUCTION_ENDED: u64 = 5;
+    const EBID_MIN_INCREMENTAL_IS_ZERO: u64 = 6;
+    const EBID_NOT_FOUND_FOR_AUCTION: u64 = 7;
+    const EONLY_AUCTION_HOUSE_OWNER_CAN_PERFORM_THIS_OPERATION: u64 = 8;
+    const EAUCTION_NOT_ENDED: u64 = 9;
+
+    struct AuctionHouseConfig has key {
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    }
+
+    struct Auctions<phantom CoinType> has key {
+        cur_auction_id: u64, // this is used to generate next auction_id
+        all_active_auctions: Table<u64, Auction<CoinType>>,
+        listing_event: EventHandle<ListingEvent>,
+        bid_event: EventHandle<BidEvent>,
+        cancel_bid_events: EventHandle<CancelBidEvent>
+    }
+
+    struct BidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct CancelBidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct Auction<phantom CoinType> has drop, store {
+        listing: Listing<CoinType>,
+        bids: vector<ID>,
+        highest_price: u64,
+        highest_bid: ID,
+        min_incremental_price: u64,
+    }
+
+    public entry fun initialize_auction_house(
+        account: &signer,
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    ) {
+        move_to(
+            account,
+            AuctionHouseConfig {
+                market_fee_denominator,
+                market_fee_numerator,
+                fee_address,
+            }
+        );
+    }
+
+    public entry fun initialize_auction<CoinType>(account: &signer) {
+        move_to(
+            account,
+            Auctions<CoinType> {
+                cur_auction_id: 0,
+                all_active_auctions: table::new(),
+                listing_event: account::new_event_handle<ListingEvent>(account),
+                bid_event: account::new_event_handle<BidEvent>(account),
+                cancel_bid_events: account::new_event_handle<CancelBidEvent>(account),
+
+            }
+        );
+    }
+
+    public fun generate_auction_data<CoinType>(
+        owner: &signer,
+        creator: address,
+        collection_name: String,
+        token_name: String,
+        property_version: u64,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
```

Comment:
do we really care?

Reply:
1. simply the auction logic so that new accepted bids are strictly higher than previous bidder.
2. participants won't be annoyed by someone just adding a very very small amount to win the auction in the last minute